sds_data_manager/lambda_code/SDSCode/query_api.py

```diff
@@ -81,19 +82,31 @@ def lambda_handler(event, context):
 
     engine = db.get_engine()
     with Session(engine) as session:
-        search_result = session.execute(query).all()
+        search_results = session.execute(query).all()
+
+    # Convert the search results (list of tuples) to a list of dicts
+    search_results = [result._asdict() for result in search_results]
+
+    # Convert datetimes to string values of format 'YYYYMMDD'
+    # Also remove values that are not needed by users
+    for result in search_results:
+        result["start_date"] = result["start_date"].strftime("%Y%m%d")
+        result["end_date"] = result["end_date"].strftime("%Y%m%d")
+        del result["id"]
+        del result["status_tracking_id"]
```

Comment:
This should be left as a follow-up PR, but out of curiosity would it make sense to:

1. store the start_date and end_date directly as strings? Do we need the datetime / does it make it handy to store that in the DB rather than just the raw strings? Above we even do conversions from the user input to dates, so there is kind of a double transformation going on here.
```python
            query = query.where(
                models.FileCatalog.start_date
                <= datetime.datetime.strptime(value, "%Y%m%d")
```
2. Have the SQL query not return `id` and `status_tracking_id` somehow?

Reply:
Good questions, and agree that they warrant a follow-up discussion/PR.

Do we know if we or the users will ever care about a time component to the start/end date? In other words, will it ever be important to know what time(s) of day a particular file covers? 

If the answer is 'yes', then it makes sense to me to keep some sort of full `datetime` component stored in the database, and then have the APIs figure out how to show that information based on the use case.